lib/widgets-page.php

```diff
@@ -41,6 +41,16 @@ function gutenberg_widgets_init( $hook ) {
 		return;
 	}
 
+	$current_screen = get_current_screen();
+	/**
+	 * Make the WP Screen object aware that this is a block editor page.
+	 * Since custom blocks check whether the screen is_block_editor,
+	 * this is required for custom blocks to be loaded and for responsive embeds
+	 * to work.
+	 * See wp_enqueue_registered_block_scripts_and_styles in wp-includes/script-loader.php
+	 */
+	$current_screen->is_block_editor( true );
```

Comment:
I copied most of the comment from the site editor:
https://github.com/WordPress/gutenberg/blob/277506eadc6ef77354c90f7984d65b243cb7c7ca/lib/edit-site-page.php#L90-L102

Two questions:
1. Is there a reason the site editor uses `global $current_screen` instead of `get_current_screen()`?
2. The comment implies this duplicates what we've implemented here, can this code now be removed? - https://github.com/WordPress/gutenberg/blob/277506eadc6ef77354c90f7984d65b243cb7c7ca/lib/widgets-page.php#L135-L143

Reply:
1. The only difference between `get_current_screen()` and `global $current_screen` is that the former will instantiate `WP_Screen` if it hasn't already been instantiated. What you did here looks good to me.

2. Yes, we should be able to remove all of that. It worked when I tried it in https://github.com/WordPress/wordpress-develop/pull/603 🙂